test/sql/copy/hive_filter_pushdown_bug.test

```diff
@@ -0,0 +1,35 @@
+# name: test/sql/copy/hive_filter_pushdown_bug.test
+# description: confirm that hive-specific filter pushdown does not mess up the filters
+# group: [copy]
+
+require parquet
+
+# Confirm
+statement ok
+COPY (SELECT i::VARCHAR as a, (i*10)::VARCHAR as b, (i*100)::VARCHAR as c from range(0,10) tbl(i)) TO '__TEST_DIR__/hive_pushdown_bug' (FORMAT PARQUET, PARTITION_BY c);
+
+query II
+explain SELECT * FROM parquet_scan('__TEST_DIR__/hive_pushdown_bug/*/*.parquet', HIVE_PARTITIONING=1) where a > b;
+----
+physical_plan	<!REGEX>:.*AND.*
+
+query II nosort q1
+explain SELECT * FROM parquet_scan('__TEST_DIR__/hive_pushdown_bug/*/*.parquet', HIVE_PARTITIONING=1) where a > b;
+----
+
+query II nosort q1
+explain SELECT * FROM parquet_scan('__TEST_DIR__/hive_pushdown_bug/*/*.parquet') where a > b;
+----
+
+query II
+explain SELECT * FROM parquet_scan('__TEST_DIR__/hive_pushdown_bug/*/*.parquet', HIVE_PARTITIONING=1) where a > c;
```

Comment:
Unrelated to this PR but I wonder if we could auto-detect the `HIVE_PARTITIONING` by default from the set of globbed files, and make the `HIVE_PARTITIONING` parameter ternary (`false, true, auto` with default on `auto`).

Reply:
Good idea! Sounds like a nice first or second issue for one of our new devs